File: ui/src/settings/infrastructure/academy.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import { ref } from 'vue';
import { defineStore } from 'pinia';

export const useAcademySettings = defineStore('academy', () => {
  const hideUnmet = ref(true);

  return {
    hideUnmet,
  };
});

File: crates/nil-core/src/world/infrastructure/building/academy.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use crate::continent::Coord;
use crate::decl_world_recruit_order_fn;
use crate::error::{Error, Result};
use crate::infrastructure::building::academy::recruit_queue::{
  AcademyRecruitOrderId,
  AcademyRecruitOrderRequest,
};
use crate::world::World;

decl_world_recruit_order_fn!(Academy);
